Server/utils/UpdateScore.js

```diff
@@ -41,11 +46,71 @@ async function UpdateScore(ID) {
 
 
     //shorthand
-    let globalScore = user.global_score;
-    let goal = user.goal;
-    let goal_bonus = 0;
+    let goal_bonus = calculateGoalBonus(user.goal);
+    let score_offset = 0; 
+    let login_bonus = 0.10; 
+    let newScores = [user.global_score, user.transport_score, user.lifestyle_score, user.diet_score, user.home_score];
+    console.log("\n\nOld Scores:\n");
+    console.log(newScores);
+    let individualEmissions = 
+        [
+            emission_entries[0].total_emissions, emission_entries[0].transport_emissions, 
+            emission_entries[0].lifestyle_emissions, emission_entries[0].diet_emissions, 
+            emission_entries[0].home_emissions    
+        ]
 
 
+    //Check if youre track for your yearly carbon emissions
+    if(emission_entries[0].total_emissions*365 <= projected_emissions) {
+        score_offset = 500*(1+login_bonus+goal_bonus);
+    }
+    else {
+        score_offset -= 300*(1-login_bonus);    
+    }
+
+    //Let each emission 
+    for(let i = 0; i < newScores.length; i++) {
+        if(i == 0) {
+            newScores[i]+=score_offset;
+        }
+        else{
+            newScores[i]+=score_offset*(1-(individualEmissions[i]/individualEmissions[0]));
+        }
+    
+        if(newScores[i] < 0) newScores[i] = 0;
+        if(newScores[i] > MAX_SCORE) newScores[i] = MAX_SCORE;
+    }
+    
+    console.log("Individual Emissions:\n");
```

Comment:
Do we need these console logs or should we delete them? (Genuine, non-sarcastic question (God I hate this even sounds sarcastic lol))